otx/algorithms/common/tasks/training_base.py

```diff
@@ -304,6 +312,80 @@ def _init_recipe_hparam(self) -> dict:
+
+                options = dict(flags=[], args={})
+                # NOTE: OTX loads image in RGB format
+                # so that `to_rgb=True` means a format change to BGR instead.
+                # Conventionally, OpenVINO IR expects a image in BGR format
+                # but OpenVINO IR under OTX assumes a image in RGB format.
+                #
+                # `to_rgb=True` -> a model was trained with images in BGR format
+                #                  and a OpenVINO IR needs to reverse input format from RGB to BGR
+                # `to_rgb=False` -> a model was trained with images in RGB format
+                #                   and a OpenVINO IR does not need to do a reverse
+                if normalize_cfg.get("to_rgb", False):
+                    options["flags"] += ["--reverse_input_channels"]
+                # value must be a list not a tuple
+                if normalize_cfg.get("mean", None) is not None:
+                    options["args"]["--mean_values"] = list(normalize_cfg.get("mean"))
+                if normalize_cfg.get("std", None) is not None:
+                    options["args"]["--scale_values"] = list(normalize_cfg.get("std"))
+
+                # fill default
+                backend_config = deploy_cfg.backend_config
+                if backend_config.get("mo_options") is None:
+                    backend_config.mo_options = ConfigDict()
+                mo_options = backend_config.mo_options
+                if mo_options.get("args") is None:
+                    mo_options.args = ConfigDict()
+                if mo_options.get("flags") is None:
+                    mo_options.flags = []
+
+                # already defiend options have higher priority
+                options["args"].update(mo_options.args)
+                mo_options.args = ConfigDict(options["args"])
+                # make sure no duplicates
+                mo_options.flags.extend(options["flags"])
+                mo_options.flags = list(set(mo_options.flags))
+
+            def patch_input_shape(deploy_cfg):
+                resize_cfg = get_configs_by_dict(
+                    self._recipe_cfg.data.test.pipeline,
+                    dict(type="Resize")
+                )
+                assert len(resize_cfg) == 1
+                resize_cfg = resize_cfg[0]
+                size = resize_cfg.size
+                if isinstance(size, int):
+                    size = (size, size)
+                assert all(isinstance(i, int) and i > 0 for i in size)
+                # default is static shape to prevent an unexpected error
+                # when converting to OpenVINO IR
+                deploy_cfg.backend_config.model_inputs = [
+                    ConfigDict(opt_shapes=ConfigDict(input=[1, 3, *size]))
+                ]
+
+            patch_input_preprocessing(deploy_cfg)
+            if not deploy_cfg.backend_config.get("model_inputs", []):
+                patch_input_shape(deploy_cfg)
+
```

Comment:
It is hard coded in each model's `deployment.py`. This logic is to patch `input shape` from `data.test.pipeline.Resize` in case it is not defined in `deployment.py`.

Reply:
I'm thinking of exposing input size options to user. This adaption might be helpful then. Thanks!